plugins/scaffolder-backend/src/scaffolder/stages/prepare/azure.ts

```diff
@@ -22,24 +22,47 @@ import { InputError, Git } from '@backstage/backend-common';
 import { PreparerBase, PreparerOptions } from './types';
 import parseGitUrl from 'git-url-parse';
 import { Config } from '@backstage/config';
+import { Logger } from 'winston';
+import {
+  readAzureIntegrationConfigs,
+  AzureIntegrationConfig,
+} from '@backstage/integration';
 
 export class AzurePreparer implements PreparerBase {
-  private readonly privateToken: string;
+  private readonly integrations: AzureIntegrationConfig[];
+  private readonly scaffolderToken: string | undefined;
 
-  constructor(config: Config) {
-    this.privateToken =
-      config.getOptionalString('scaffolder.azure.api.token') ?? '';
+  constructor(config: Config, { logger }: { logger: Logger }) {
```

Comment:
Think it would be a bit nicer to have a `AzurePreparer.fromConfig` rather than doing all of this work in the constructor.

Reply:
i've been liking the pattern of `constructor(private readonly opts: { x: string; y: integer... }) {}` so you get all of your settings in one go :) no need to line by line copy the N constructor arguments to fields